sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -2038,3 +2041,16 @@ def test_neighbors_distance_metric_deprecation():
 def test_valid_metrics_has_no_duplicate():
     for val in neighbors.VALID_METRICS.values():
         assert len(val) == len(set(val))
+
+
+def test_regressor_predict_on_arraylikes():
+    """Non-regression test for #22687"""
```

Comment:
please begin by a small description on the purpose of this test

Reply:
Updated description, thanks!